### PR TITLE
Adjust recipe and exclude py38 using a migrator

### PR DESCRIPTION
**Pull request by mariusvniekerk**

### CODE REVIEW

File: .ci_support/migrations/python311.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+migrator_ts: 1666686085
```
Comment (mariusvniekerk) [inline .ci_support/migrations/python311.yaml:1]: Just pre-adding this one, since we probably want it in any case

File: recipe/meta.yaml [hunk 2]
```diff
@@ -153,18 +153,17 @@ outputs:
...
         - {{ compiler('cxx') }}
-        - cross-python_{{ target_platform }}                                     # [build_platform != target_platform]
-        - cython
-        - pip
-        - python >=3.9
-        - setuptools
+        - python                                 # [build_platform != target_platform]
+        - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+        - cython                                 # [build_platform != target_platform]
+        - numpy                                  # [build_platform != target_platform]
```
Comment (lidavidm) [inline recipe/meta.yaml:159]: Hmm, why do we need numpy?

File: .ci_support/migrations/remove-python38.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+__migrator:
```
Comment (mariusvniekerk) [inline .ci_support/migrations/remove-python38.yaml:1]: This is a trick that we can use to remove a particular variant key from the expanded set

**Review by lidavidm — APPROVED** [review]
Thanks!

**Pull request merged**